### DISCUSSION TITLE
Backdating Purchase Order Recipts for Accurate Stock History

### DISCUSSION THREAD

**Discussion by George-Stoker** [OCA/odoo-community.org#159] [Q&A] [answered]
Hi all, I'm currently trying to import my historical inventory data from my old ERP to Odoo 16 chronologically using XMLRPC. I am trying to emulate the way you would do this on the frontend as much as possible (same actions ect) so I don't miss some hidden fields in other models and mess up the dashboard and other analytical features. My problem is, using button_validate always sets my date_done to the current timestamp even if I try to write a date_done value before using it. I also tried to change the date_done after validation but this doesn't work as it is a read only value and even if I were to remove this protection, this edit would only changed the current record and not the other related models. 

Is there a standard work around for this in, preferably default, Odoo? If not, does anyone have any suggestions that would let me keep the functionality.

**Comment by francesco-ooops** [comment]
Hi, I think this module (not available in recent versions) should work as
requested: https://github.com/OCA/stock-logistics-workflow/pull/1494

Francesco

Il giorno gio 21 mar 2024 alle ore 17:31 MyBrainIsSmall <
***@***.***> ha scritto:

> Hi all, I'm currently trying to import my historical inventory data from
> my old ERP to Odoo 16 chronologically using XMLRPC. I am trying to emulate
> the way you would do this on the frontend as much as possible (same actions
> ect) so I don't miss some hidden fields in other models and mess up the
> dashboard and other analytical features. My problem is, using
> button_validate always sets my date_done to the current timestamp even if I
> try to write a date_done value before using it. I also tried to change the
> date_done after validation but this doesn't work as it is a read only value
> and even if I were to remove this protection, this edit would only changed
> the current record and not the other related models.
>
> Is there a standard work around for this in, preferably default, Odoo? If
> not, does anyone have any suggestions that would let me keep the
> functionality.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/OCA/discussions/159>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANHLKUPZWJRKQ4AORPYB4W3YZMDPLAVCNFSM6AAAAABFBZQI4CVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGQYDKMBTGI>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 

*Francesco Foresti*
*Sicurpharma Srl*
***@***.***
+39 333 8123 790

**Comment by antoniospneto** [comment, marked as answer]
If you know well what you're doing, you can try to directly include the old records into the database using SQL, which allows you to bypass some of the validations and standard process of the Odoo ORM.

**Comment by yostashiro** [comment]
There is this PR which serves a similar purpose as https://github.com/OCA/stock-logistics-workflow/pull/1494:

[16.0][ADD] stock_picking_accounting_date #1711
https://github.com/OCA/stock-logistics-warehouse/pull/1711

You can also try this out:

stock_valuation_layer_accounting_date
https://github.com/OCA/stock-logistics-warehouse/tree/16.0/stock_valuation_layer_accounting_date
